controllers/controllers.go

```diff
@@ -16,7 +16,49 @@
 
 package controllers
 
-import "errors"
+import (
+	"errors"
+
+	"github.com/openshift-kni/numaresources-operator/pkg/status"
+)
+
+type conditionInfo struct {
+	Type    string // like metav1.Condition.Type
+	Reason  string
+	Message string
+}
+
+func availableConditionInfo() conditionInfo {
+	return conditionInfo{
+		Type:   status.ConditionAvailable,
+		Reason: "AsExpected",
+	}
+}
+
+func progressingConditionInfo(reason, message string) conditionInfo {
+	return conditionInfo{
+		Type:    status.ConditionProgressing,
+		Message: message,
+		Reason:  reason,
+	}
+}
+
+func degradedConditionInfoFromError(err error) conditionInfo {
+	return conditionInfo{
+		Type:    status.ConditionDegraded,
+		Message: messageFromError(err),
+	}
```

Comment:
should we add reason here? 
`Reason: reasonFromError(err)`

Reply:
we probably should, yes. It should be helpful and harmless.